webpack.common.js

```diff
@@ -1,15 +1,31 @@
 const path = require('path')
+const fs = require('fs')
+
 const HTMLWebpackPlugin = require('html-webpack-plugin')
 const FaviconsWebpackPlugin = require('favicons-webpack-plugin')
 
+const localConfig = path.resolve(__dirname, 'config-local.js')
+const defaultConfig = path.resolve(__dirname, 'config.js')
+let config
+if (fs.existsSync(localConfig)) {
+    config = localConfig
+} else if (fs.existsSync(defaultConfig)) {
+    config = defaultConfig
+} else {
+    throw `The config file is missing: ${defaultConfig}`
+}
+
```

Comment:
In #120 there was an additional check that made sure that config includes a gh key, but I'm not sure if this is needed. If there is no api key we currently get a 401 response for graphhopper.com/api/1/info and if we handle this properly we could show a nice 'missing api key' message right in the UI. But maybe we still need to deal with missing tile keys (unless we just use OSM by default, see above). Not checking for the GH key here has the advantage that it is not needed anyway when we set the url to localhost (using a local GH server).

Reply:
Why do we need two config files?

Reply:
Ok, sorry. Just read your main comment... it is clear now.

Reply:
Can we make OSM tiles the default (without api key)? Or maybe omniscale like in maps 1.0? It seems kind of ugly to tell everyone to get a maptiler key, but not sure if using OSM without key is ok.

Reply:
Currently raster tiles would be ugly due to the flickering issue. In general we can do this, sure. (for production I would still keep using maptiler or mapilion)

Reply:
> not sure if using OSM without key is ok.

They do not have any API key protection

Reply:
The problem seems to be that we get a CORS error when we request /info with a wrong API key :(

![image](https://user-images.githubusercontent.com/17603532/125494886-3112d121-6a2e-4b44-abc1-4c58ed089181.png)

This makes it hard to show a nice error message.

Reply:
I mean sure we could already check if there is an api key in webpack but a) this check will fail unecessarily when we use localhost and b) even when the api key exists it could still be wrong and we'd like to see a useful error message in the url. Nothing major, but would be nice to have.

Reply:
This is an older known issue behind the scenes which we could/should fix

Reply:
> Currently raster tiles would be ugly due to the flickering issue. In general we can do this, sure. (for production I would still keep using maptiler or mapilion)

Yes, I mean we could use OSM by default (in `config.js`) and then in the production config (`config-local.js` that we inject via travis) we set it to map tiler.

> They do not have any API key protection

Yes, I was just wondering if it is ok to use, but actually I don't see a reason not to use it as default when we can already use it via the layers menu.

> This is an older known issue behind the scenes which we could/should fix

Ok, I'll open a separate issue for this.